From 346aed7fb262bf5d46628d43785841fbfb31c5e7 Mon Sep 17 00:00:00 2001
From: Markus Boehme <markus.boehme@chainguard.dev>
Date: Tue, 9 Dec 2025 21:25:04 +0000
Subject: [PATCH 2/5] vendor: sha1cd: do not register as crypto hash
 implementation

In the vendored github.com/pjbgf/sha1cd package, do not register the
sha1cd implementation as a SHA1 hash with the standard library's crypto
package to require uses of this SHA1 implementation to be individually
acknowledged.

Signed-off-by: Markus Boehme <markus.boehme@chainguard.dev>
---
 vendor/github.com/pjbgf/sha1cd/sha1cd.go | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/vendor/github.com/pjbgf/sha1cd/sha1cd.go b/vendor/github.com/pjbgf/sha1cd/sha1cd.go
index b8d2890a..865a995c 100644
--- a/vendor/github.com/pjbgf/sha1cd/sha1cd.go
+++ b/vendor/github.com/pjbgf/sha1cd/sha1cd.go
@@ -12,7 +12,6 @@ package sha1cd
 // Original: https://github.com/golang/go/blob/master/src/crypto/sha1/sha1.go
 
 import (
-	"crypto"
 	"encoding/binary"
 	"errors"
 	"hash"
@@ -20,10 +19,6 @@ import (
 	shared "github.com/pjbgf/sha1cd/internal"
 )
 
-func init() {
-	crypto.RegisterHash(crypto.SHA1, New)
-}
-
 // The size of a SHA-1 checksum in bytes.
 const Size = shared.Size
 
-- 
2.51.2

